plugin/core/registry.py

```diff
@@ -144,11 +143,31 @@ def client_with_capability(self, capability: str) -> Optional[Client]:
         return client_from_session(session_for_view(self.view, capability))
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
+class LspRestartAllServersCommand(sublime_plugin.TextCommand):
     def is_enabled(self) -> bool:
         return is_supported_view(self.view)
 
     def run(self, edit: Any) -> None:
         window = self.view.window()
         if window:
             windows.lookup(window).restart_sessions()
+
+
+class LspRestartServerCommand(sublime_plugin.TextCommand):
+    def is_enabled(self) -> bool:
+        return is_supported_view(self.view) and windows.lookup(self.view.window()).has_active_sessions()
+
+    def run(self, edit: Any) -> None:
+        window = self.view.window()
+        if not window:
+            return
+
+        sessions = windows.lookup(window).get_sessions()
+        if sessions:
+            window.show_quick_panel(sessions, lambda index: self.restart_server(index, sessions))
```

Comment:
Maybe skip the quick panel if there is only one server running?

Reply:
Possibly, it would make the UX a bit smoother.

Reply:
Suggestion:
```python
        if len(config_names) == 1:
            self.restart_server(window, 0, config_names)
        elif config_names:
            window.show_quick_panel(config_names, lambda index: self.restart_server(window, index, config_names))
```